light-clients/ics10-grandpa-cw/src/contract.rs

```diff
@@ -134,12 +134,7 @@ pub fn execute(
 	let client = GrandpaClient::<HostFunctions>::default();
 	let mut ctx = Context::<HostFunctions>::new(deps, env);
 	let client_id = ClientId::from_str("08-wasm-0").expect("client id is valid");
-	let result = process_message(msg, client, &mut ctx, client_id.clone());
-	let data = match result {
-		Ok(res) => res,
-		Err(ContractError::Grandpa(e)) => to_binary(&ContractResult::error(e))?,
-		Err(e) => return Err(e),
-	};
+	let data = process_message(msg, client, &mut ctx, client_id.clone())?;
```

Comment:
We need to test this because the 08-wasm module might not expect a grandpa error

Reply:
Seems like it does as `call()` returns `(ContractResult, error)`

```go
// call calls the contract with the given payload and writes the result to output.
func call[T ContractResult](ctx sdk.Context, clientStore sdk.KVStore, cs *ClientState, payload any) (T, error) {
...
}
```

Reply:
Ran the tests on this new tendermint-cw on the ibc-go branch and tests pass